### ISSUE TITLE
New POD fields are not syncing correctly with new GF fields since 1.4.1 update

### ISSUE THREAD

**Issue by ChrisO2134** [pods-framework/pods-gravity-forms#125]
## Issue Overview
Something has gone seriously wrong with Pods and Gravity forms?? Existing fields in forms can still see and update pods fields, but no new fields in forms or pods will talk to each other. Even worse than that my new fields in forms are just pulling random field data from the pod to display in the form?? For example, I have just created a field in Pods 'User Telephone' and a field in the form 'Telephone' , but after setting them up to talk to each other the 'Telephone' field on the form is displaying the 'User Bio' which is an already existing field further down the form. How can this be happening? This is not a one off. I have tried on several forms and have similar issues. All new fields are corrupted in some way while all previously existing fields work. This is only affecting form and Pod fields created after the 1.4.1 update. Previously to that I could create new pod fields/form fields easily and then sync fine

Screenshots attached. Post excerpt is an old form field, that works fine and syncs with the post excerpt of the POD. 
The 'merch_action_notification' is a new POD field that is set up to sync with the Action Required GF form field. BUT somehow the new form field 'Action Required' is pulling the Post Excerpt from the POD? 
Editing the Action Required on the form and submitting the data does nothing to either the Excerpt , or the merch_action_notification. Something is affecting the communication between the new fields ? 

I can't include link to live example as I have reverted to 1.3, which works brilliantly. I have tried this with other plugins disabled, and it still creates the error.

Hope this is something that can be reproduced, or a possible solution found. My whole site is based on GF and PODS working together.

![2018-10-23 1](https://user-images.githubusercontent.com/29980881/47340065-0a88aa80-d695-11e8-8730-d2a301ad826b.png)
![2018-10-23](https://user-images.githubusercontent.com/29980881/47340066-0a88aa80-d695-11e8-9f53-287b020dd13b.png)

## Possible Solution
Revert the options on the PODs tab in GF back to the last version

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Form Upload

<details>
```
Upload Gravity Form JSON or Copy and Paste it Here.
```
</details>

## Pods Package Export

<details>
```
Copy and Paste the JSON Export from Pods Migrate: Packages, Export and Paste it Here
```
</details>

## Todos
- [ ] Tests
- [ ] Documentation

## WordPress Environment
<details>
```
Debug Information
WordPress Version: 4.9.8

PHP Version: 5.6.29

MySQL Version: 5.6.41

Server Software: Apache

Your User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.67 Safari/537.36

Session Save Path: /tmp

Session Save Path Exists: Yes

Session Save Path Writeable: Yes

Session Max Lifetime: 1440

Opcode Cache:

Apc: No
Memcached: No
OPcache: No
Redis: No
Object Cache:

APC: No
APCu: No
Memcache: No
Memcached: No
Redis: No
WPDB Prefix: oifh_

WP Multisite Mode: No

WP Memory Limit: 40M

Pods Network-Wide Activated: No

Pods Install Location: /home/zoncoupo/public_html/uk/wp-content/plugins/pods/

Pods Tableless Mode Activated: No

Pods Light Mode Activated: No

Currently Active Theme: 2017 Child

Currently Active Plugins:

(Yes I have tried this with other plugins disabled, and it still creates the error)

Amazon Associates Link Builder: 1.8.0
Delete Me: 2.7
Duplicate Post: 3.2.2
Email Marketing by Drip: 1.0.2
Gravity Forms: 2.3.5
Gravity Forms Drip Add-On: 2.1.0
Gravity Forms User Registration Add-On: 3.9
GravityView: 2.1.0.3
GravityView - Advanced Filter Extension: 1.3
Limit Login Attempts: 1.7.1
Log Out Shortcode: 1.0.5
Pods - Custom Content Types and Fields: 2.7.9
Pods Gravity Forms Add-On: 1.3
Post Expirator: 2.3.1.1
Private content: 4.4
Really Simple SSL: 3.1.1
Say What?: 1.9.1
Search & Filter Pro: 2.3.2
Simple Custom CSS: 4.0.1
URL Params: 2.1
User Shortcodes Plus: 2.0.1
Widget Shortcode: 0.3.3
WP File Manager: 3.2
WP User Manager: 2.0.7

```
</details>

**Comment by sc0ttkclark** [comment]
I recommend downgrading back down to the version you had before until we figure out what's going on here.

https://downloads.wordpress.org/plugin/pods-gravity-forms.1.3.zip

I'll take a look at this to see what might cause this. But I do have a few questions:

1. Do you have multiple Pods form feeds for this form? 
2. Do you have multiple forms that have a Pods feed to send data to Pods?

**Comment by ChrisO2134** [comment]
Hi Scott, 
Yes I have downgraded which is why I can't send a link to the form now. The 1.3 version still works perfectly. 

To answer your questions... 

1 - No , only 1 feed per form. (I tried to do multiple feeds a few months ago without success, so have built the website only using 1 feed per form after that.)

2 - Yes I have many forms. Approx 50 across the website, with about half of them running pod feeds. Most of them only update 1 or 2 fields per pod at any one time. 

Thanks 
Chris

**Comment by sc0ttkclark** [comment]
And you were for sure running Pods 1.4.1 and not Pods 1.4? There was a brief bug in 1.4 that was patched in 1.4.1 where there were cache collisions in the field data being retrieved from GF for the Pods feed which may have been what this was related to.

**Comment by ChrisO2134** [comment]
Yes I think so. When I started noticing these errors I checked I had the most recent plugin versions. Nothing was showing as 'update required', but to be honest I cannot recall if it was showing as 1.4 or 1.4.1 

I will upgrade again this eve and do some testing to confirm it was 1,4,1. It may take a while before I get a chance to double check this version and report back later.

**Comment by sc0ttkclark** [comment]
No problem. I'll take a deeper look at the code and see if there's anything else I can find too.

**Comment by ChrisO2134** [comment]
Ok , so I have just reinstalled the 1.4.1 and immediately have replicated the 'No Data' being synced between the newest fields on PODS or FORMS. So far I have not replicated getting the 'Wrong Data' being synced, but I think it will happen if I keep trying to update fields. 
Anyway , just a heads up that the initial problem I was having with new fields is still there. 
I can still sync ok with fields created before the initial update several days ago. But any new fields created since are not playing nicely. 

I have downgraded again to 1.3 and immediately everything is back to full working order. 

Not much to go on maybe, but hope this has helped in some way. 

Cheers

**Comment by sc0ttkclark** [comment]
Gotcha, I'll follow up as soon as I have a fix ready.

**Comment by telarid** [comment]
Hi there I am finding this while updating my posts. Is there an update to this issue yet.
Thank you for your time.

Cheers

**Comment by secondsky** [comment]
Well, i did observe the same thing.
Out of nothing, if new fields have been created in Pods and GF, these fields dont get populated anymore from GF to Pods post. Unfortunately i have no clue what caused this as it worked to some point perfectly, even new fields have been added in the past. But than, out of a sudden, for new fields it stopped to work... And as since 2018 this is open, i guess there will be no solution until 2030 :-)

**Comment by sc0ttkclark** [comment]
I'll be looking at this issue this month, sorry for the delay.

**Comment by sc0ttkclark** [comment]
I haven't been able to reproduce this but I am releasing Pods Gravity Forms 1.4.3  with some important fixes that I did find issues with. We'll keep looking into this issue to see if we can reproduce it.

**Comment by jimtrue** [comment]
@secondsky Can you please provide the JSON export from Pods Migrate: Packages, Export and Paste it Here in your reply (or save it as a text file and upload). We only need to see the structure of the Pod that you know you have an issue with and we need an export of the form that you have an issue with as well so we can replicate the problem, especially since none of us have been able to replicate the problem.

We're happy to help you, but we need those details because right now, we only know you have an issue, but nothing to test with.

**Comment by sc0ttkclark** [comment]
@ChrisO2134 @telarid If either of you have any packages/further details like @jimtrue was asking for, that would greatly help us track this down. Thanks!

**Comment by jimtrue** [comment]
@secondsky and @telarid There was some breaking updates in 1.4.2, which are definitely fixed in 1.4.3. They do require patching to Pods 2.7.17 as well, so plan on making a backup of your configuration and testing, but I think you'll be pleased with Pods 2.7.17 and Pods GF 1.4.3, I know I was!
